src/middleware/auth/logto.ts

```diff
@@ -50,6 +49,135 @@ export class LogToHelper {
     return this.allResourceWithNames
   }
 
+  public async setDefaultRoleForUser(userId: string): Promise<ILogToErrorResponse | void> {
+    const roles = await this.getRolesForUser(userId)
+    if (roles && roles.status === 200 && roles.data.length === 0) {
+        const default_role = await this.getDefaultRole()
+        if (default_role && default_role.status === 200) {
+            return await this.assignDefaultRoleForUser(userId, default_role.data.id)
+        }
+    }
+  }
+
+  private async assignDefaultRoleForUser(userId: string, roleId: string): Promise<ILogToErrorResponse | void> {
+    const uri = new URL(`/api/roles/${roleId}/users`, process.env.LOGTO_ENDPOINT);
+    try {
+        const body = {
+            userIds: [userId],
+        };
+        const response = await fetch(uri, {
+            headers: {
+                Authorization: 'Bearer ' + this.m2mToken,
+                'Content-Type': 'application/json',
+            },
+            method: 'POST',
+            body: JSON.stringify(body),
+        });
+
+        if (response.ok) {
+            
+            return {
+                status: 200,
+                error: "",
+                data: {}
+            }
+        }
+        return {
+            error: await response.text(),
+            status: response.status,
+            data: {}
+        }
+
+        
+    } catch (err) {
+        return {
+            error: `getRolesForUser ${err}`,
+            status: 500,
+            data: {}
+        }
+    }
+  }
+
+  private async getDefaultRole(): Promise<ILogToErrorResponse | void> {
+    const uri = new URL(`/api/roles`, process.env.LOGTO_ENDPOINT);
+    try {
+        const response = await fetch(uri, {
+            headers: {
+                Authorization: 'Bearer ' + this.m2mToken,
+            },
+        });
+
+        if (response.ok) {
+
+            const metadata = await response.json()
+
+            for (const role of metadata) {
+                if (role.name === process.env.LOGTO_DEFAULT_ROLE) {
+                    return {
+                        status: 200,
+                        error: "",
+                        data: role
+                    }
+                }
+            }
+            
+            return {
+                status: 200,
+                error: "",
+                data: metadata
+            }
+        }
+        return {
+            error: await response.text(),
+            status: response.status,
+            data: {}
+        }
+
+        
+    } catch (err) {
+        return {
+            error: `getRolesForUser ${err}`,
+            status: 500,
+            data: {}
+        }
+    }
+  }
```

Comment:
Honestly this is a more complicated way of doing things. Ask for the role ID, then it's a simple role ID GET. https://docs.logto.io/api/#tag/Roles/paths/~1api~1roles~1:id/get

Try to match a name is an inexact science and gets into the realm of getting case of the string right or wrong etc.